gammapy/irf/psf_table.py

```diff
@@ -502,26 +503,66 @@ def write(self, *args, **kwargs):
         """
         self.to_fits().writeto(*args, **kwargs)
 
-    def table_psf_at_energy(self, energy, **kwargs):
-        """TablePSF at a given energy.
+    def evaluate(self, energy=None, offset=None,
+                 interp_kwargs=None):
+        """Interpolate the value of the `EnergyOffsetArray` at a given offset and Energy.
 
-        Extra ``kwargs`` are passed to the `~gammapy.irf.TablePSF` constructor.
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy value
+        offset : `~astropy.coordinates.Angle`
+            offset value
+        interp_kwargs : dict
+            option for interpolation for `~scipy.interpolate.RegularGridInterpolator`
+
+        Returns
+        -------
+        values : `~astropy.units.Quantity`
+            Interpolated value
+        """
+        if not interp_kwargs:
+            interp_kwargs = dict(bounds_error=False, fill_value=None)
+
+        from scipy.interpolate import RegularGridInterpolator
+        if energy is None:
+            energy = self.energy
+        if offset is None:
+            offset = self.offset
+
+        energy = Energy(energy).to('TeV')
+        offset = Angle(offset).to('deg')
+
+        energy_bin = self.energy.to('TeV')
```

Comment:
@adonath 
here compare to the ```EnergyOffsetArray``` class, you directly store the center of the bin in offset and energy!